examples/undocumented/python/graphical/classifier_gaussian_process_binary_classification.py

```diff
@@ -1,61 +1,52 @@
+import itertools
 
-from pylab import *
-from numpy import *
-from itertools import *
+import matplotlib.pyplot as plt
+import numpy as np
 
-def generate_toy_data(n_train=100, mean_a=asarray([0, 0]), std_dev_a=1.0, mean_b=3, std_dev_b=0.5):
 
+def generate_toy_data(n_train=100, mean_a=np.asarray([0, 0]), std_dev_a=1.0, mean_b=3, std_dev_b=0.5):
     # positive examples are distributed normally
-    X1 = (random.randn(n_train, 2)*std_dev_a+mean_a).T
+    X1 = (np.random.randn(n_train, 2) * std_dev_a + mean_a).T
 
     # negative examples have a "ring"-like form
-    r = random.randn(n_train)*std_dev_b+mean_b
-    angle = random.randn(n_train)*2*pi
-    X2 = array([r*cos(angle)+mean_a[0], r*sin(angle)+mean_a[1]])
+    r = np.random.randn(n_train) * std_dev_b + mean_b
+    angle = np.random.randn(n_train) * 2 * np.pi
+    X2 = np.array([r * np.cos(angle) + mean_a[0], r * np.sin(angle) + mean_a[1]])
 
     # stack positive and negative examples in a single array
-    X_train = hstack((X1,X2))
+    X_train = np.hstack((X1, X2))
 
     # label positive examples with +1, negative with -1
-    y_train = zeros(n_train*2)
+    y_train = np.zeros(n_train * 2)
     y_train[:n_train] = 1
     y_train[n_train:] = -1
 
     return [X_train, y_train]
 
-def gaussian_process_binary_classification_laplace(X_train, y_train, n_test=50):
 
-    # import all necessary modules from Shogun (some of them require Eigen3)
-    try:
-        from shogun import RealFeatures, BinaryLabels, GaussianKernel, \
-            LogitLikelihood, ProbitLikelihood, ZeroMean, SingleLaplacianInferenceMethod, \
-            EPInferenceMethod, GaussianProcessClassification
-    except ImportError:
-        print('Eigen3 needed for Gaussian Processes')
-        return
+def gaussian_process_binary_classification_laplace(X_train, y_train, n_test=50):
     import shogun as sg
-    import numpy as np
 
     # convert training data into Shogun representation
-    train_features = RealFeatures(X_train)
-    train_labels = BinaryLabels(y_train)
+    train_features = sg.features(X_train)
+    train_labels = sg.BinaryLabels(y_train)
 
     # generate all pairs in 2d range of testing data
-    x1 = linspace(X_train[0,:].min()-1, X_train[0,:].max()+1, n_test)
-    x2 = linspace(X_train[1,:].min()-1, X_train[1,:].max()+1, n_test)
-    X_test = asarray(list(product(x1, x2))).T
+    x1 = np.linspace(X_train[0, :].min() - 1, X_train[0, :].max() + 1, n_test)
+    x2 = np.linspace(X_train[1, :].min() - 1, X_train[1, :].max() + 1, n_test)
+    X_test = np.asarray(list(itertools.product(x1, x2))).T
 
     # convert testing features into Shogun representation
-    test_features = RealFeatures(X_test)
+    test_features = sg.features(X_test)
 
     # create Gaussian kernel with width = 2.0
-    kernel = sg.kernel("GaussianKernel", log_width=np.log(2.0))
+    kernel = sg.kernel('GaussianKernel', log_width=np.log(2.0))
 
     # create zero mean function
-    mean = ZeroMean()
+    mean = sg.ZeroMean()
```

Comment:
is there a factory for mean functions? I think so as we have a gp meta example

Reply:
I did not find.